core-rust/state-manager/src/store/rocks_db.rs

```diff
@@ -764,6 +800,10 @@ impl ActualStateManagerDatabase {
         state_manager_database.restore_december_2023_lost_substates(network);
         state_manager_database.ensure_historical_substate_values();
 
+        if state_manager_database.config.enable_entity_listing_indices {
```

Comment:
It would be more consistent to use an `ensure_` style and have this config check internal to the method? To either set up or tear down the index.

Reply:
We're also missing the teardown logic for this index.